File: Julia/problem003.jl
#=
Problem 3 - Project Euler
http://projecteuler.net/index.php?section=problems&id=3
=#

function maxprimefactor(x)
    divider = 2
    target = x
    while target != 1
        if target % divider == 0
            target /= divider
        else
            divider += 1
        end
    end
    divider
end

println(maxprimefactor(600851475143))

File: Kotlin/src/main/kotlin/problem003.kt
/*
 * Problem 3 - Project Euler
 * http://projecteuler.net/index.php?section=problems&id=3
 */
fun maxprimefactor(x: Long): Long {
  var divider: Long = 2
  var target: Long = x

  while (target != 1L) {
    if (target % divider == 0L) {
      target /= divider
    } else {
      divider += 1
    }
  }
  return divider
}

fun main() {
  println(maxprimefactor(600851475143))
}
